# Load Libraries


In [297]:
library("rcompanion")
library("car")
library("effects")
library("multcomp")
library("dplyr")

# Load in Data


In [298]:
NYR=read.csv("/Users/jeannekiluva/Desktop/New_years_resolutions.csv")

In [299]:
head(NYR)

tweet_created      
1 2014-12-21 16:11:00
2 2014-12-21 16:16:00
3 2014-12-21 16:30:00
4 2014-12-21 17:51:00
5 2014-12-21 18:09:00
6 2014-12-21 18:18:00
  tweet_text                                                                                                                                      
1 #NewYearsResolution to not put the parking lot ticket directly in my mouth right after getting it from the machine #ew                          
2 I got the hair. I just need to work on the body to pull off Lara Croft next Halloween #NewYearsResolution http://t.co/Ihqh5Z7prx                
3 Self improvement!.. Mentally, physically, and financially. #NewYearsResolution                                                                  
4 I've decided to get pregnant in #2015 #NewYearsResolution                                                                                       
5 All I want for Christmas is to be Carrie Underwood. #NewYearsResolution                                                                         
6 Note to self: "Stop being lazy &amp; get back into hot yoga." I have a monthly pass, so no excuses! #hotyoga #newyearsresolution #healthiswealth
  tweet_category   tweet_topics                                         
1 Humor            Humor about Personal Growth and Interests Resolutions
2 Health & Fitness Improve my body                                      
3 Personal Growth  Improve my attitude                                  
4 Humor            Humor about Family/Friends/Relationships Resolutions 
5 Personal Growth  Other                                                
6 Health & Fitness Excersize More                                       
  tweet_location         tweet_state tweet_region user_timezone             
1 City of Angels, CA     CA          West         Pacific Time (US & Canada)
2 Arlington, TX          TX          South        Central Time (US & Canada)
3 Las Vegas, NV          NV          West         Central Time (US & Canada)
4 Ft. Luderdale, Florida FL          South                                  
5 South Carolina         SC          South        Pacific Time (US & Canada)
6 NYC                    NY          Northeast    Quito                     
  user_gender retweet_count
1 male        NA           
2 male        NA           
3 male        NA           
4 male        NA           
5 female      NA           
6 female      NA

# Rename Columns

In [300]:
NYR <- NYR%>% rename(Date=tweet_created,Tweet=tweet_text,Category=tweet_category,Topics=tweet_topics,Location=tweet_location,State=tweet_state,Region=tweet_region,TZ=user_timezone,Gender=user_gender)

In [301]:
head(NYR)

Date               
1 2014-12-21 16:11:00
2 2014-12-21 16:16:00
3 2014-12-21 16:30:00
4 2014-12-21 17:51:00
5 2014-12-21 18:09:00
6 2014-12-21 18:18:00
  Tweet                                                                                                                                           
1 #NewYearsResolution to not put the parking lot ticket directly in my mouth right after getting it from the machine #ew                          
2 I got the hair. I just need to work on the body to pull off Lara Croft next Halloween #NewYearsResolution http://t.co/Ihqh5Z7prx                
3 Self improvement!.. Mentally, physically, and financially. #NewYearsResolution                                                                  
4 I've decided to get pregnant in #2015 #NewYearsResolution                                                                                       
5 All I want for Christmas is to be Carrie Underwood. #NewYearsResolution                                                                         
6 Note to self: "Stop being lazy &amp; get back into hot yoga." I have a monthly pass, so no excuses! #hotyoga #newyearsresolution #healthiswealth
  Category         Topics                                               
1 Humor            Humor about Personal Growth and Interests Resolutions
2 Health & Fitness Improve my body                                      
3 Personal Growth  Improve my attitude                                  
4 Humor            Humor about Family/Friends/Relationships Resolutions 
5 Personal Growth  Other                                                
6 Health & Fitness Excersize More                                       
  Location               State Region    TZ                         Gender
1 City of Angels, CA     CA    West      Pacific Time (US & Canada) male  
2 Arlington, TX          TX    South     Central Time (US & Canada) male  
3 Las Vegas, NV          NV    West      Central Time (US & Canada) male  
4 Ft. Luderdale, Florida FL    South                                male  
5 South Carolina         SC    South     Pacific Time (US & Canada) female
6 NYC                    NY    Northeast Quito                      female
  retweet_count
1 NA           
2 NA           
3 NA           
4 NA           
5 NA           
6 NA

#Drop Columns

In [302]:
NYRtrim = (within(NYR, rm(Location,retweet_count,TZ)))

In [303]:
na.omit(NYRtrim)

Date               
1    2014-12-21 16:11:00
2    2014-12-21 16:16:00
3    2014-12-21 16:30:00
4    2014-12-21 17:51:00
5    2014-12-21 18:09:00
6    2014-12-21 18:18:00
7    2014-12-21 18:50:00
8    2014-12-21 19:12:00
9    2014-12-21 19:32:00
10   2014-12-21 19:46:00
11   2014-12-21 19:46:00
12   2014-12-21 20:05:00
13   2014-12-21 20:10:00
14   2014-12-21 20:37:00
15   2014-12-21 20:51:00
16   2014-12-21 21:03:00
17   2014-12-21 21:09:00
18   2014-12-21 21:23:00
19   2014-12-21 21:53:00
20   2014-12-21 22:32:00
21   2014-12-21 22:50:00
22   2014-12-21 23:22:00
23   2014-12-22 02:09:00
24   2014-12-22 02:57:00
25   2014-12-22 03:25:00
26   2014-12-22 04:25:00
27   2014-12-22 05:20:00
28   2014-12-22 05:20:00
29   2014-12-22 07:43:00
30   2014-12-22 07:55:00
⋮    ⋮                  
4694 2015-01-02 09:42:00
4695 2015-01-02 09:42:00
4696 2015-01-02 09:42:00
4697 2015-01-02 09:42:00
4698 2015-01-02 09:43:00
4699 2015-01-02 09:43:00
4700 2015-01-02 09:44:00
4701 2015-01-02 09:44:00
4702 2015-01-02 09:44:00
4703 2015-01-02 09:44:00
4704 2015-01-02 09:44:00
4705 2015-01-02 09:44:00
4706 2015-01-02 09:44:00
4707 2015-01-02 09:45:00
4708 2015-01-02 09:46:00
4709 2015-01-02 09:46:00
4710 2015-01-02 09:46:00
4711 2015-01-02 09:47:00
4712 2015-01-02 09:47:00
4713 2015-01-02 09:48:00
4714 2015-01-02 09:49:00
4715 2015-01-02 09:49:00
4716 2015-01-02 09:49:00
4717 2015-01-02 09:51:00
4718 2015-01-02 09:51:00
4719 2015-01-02 09:52:00
4720 2015-01-02 09:52:00
4721 2015-01-02 09:53:00
4722 2015-01-02 09:54:00
4723 2015-01-02 09:54:00
     Tweet                                                                                                                                                    
1    #NewYearsResolution to not put the parking lot ticket directly in my mouth right after getting it from the machine #ew                                   
2    I got the hair. I just need to work on the body to pull off Lara Croft next Halloween #NewYearsResolution http://t.co/Ihqh5Z7prx                         
3    Self improvement!.. Mentally, physically, and financially. #NewYearsResolution                                                                           
4    I've decided to get pregnant in #2015 #NewYearsResolution                                                                                                
5    All I want for Christmas is to be Carrie Underwood. #NewYearsResolution                                                                                  
6    Note to self: "Stop being lazy &amp; get back into hot yoga." I have a monthly pass, so no excuses! #hotyoga #newyearsresolution #healthiswealth         
7    RT @ashley_pg19: I'm sick of waiting on life to happen. I want to make my life worth living. _Ù\u008dâ_ÙÎÈ #life #loveyourlife #christmastime #NewYear‰Û_
8    I have to go do my 6 mile run. #stayingFIT #toneUP #StallionBOOTY #NewYearsResolution starts now.                                                        
9    social media purging &gt; #NewYearsResolution                                                                                                            
10   My #NewYearsResolution is to gain a bunch of weight, then lose it                                                                                        
11   ‰ÛÏ@PartiPants27: My #NewYearsResolution is to make "dem gainz" By any means necessary‰Û\u009d bang bang muscle gainz                                    
12   My #NewYearsResolution should be to make payments on my student loan that are equal to my @ULTA_Beauty orders. #platinum                                 
13   #NewYearsResolution 1: Acknowledge life's  futility by becoming an alcoholic.  (All tips and advice are welcome)                                         
14   they gotta listen tho "@K1NG_JAMES_ #NewYearsResolution teach everyone I come in contact with a positive lesson that helps them out in life"             
15   RT @K1NG_JAMES_: I just wanna be as happy as the ppl that be on m

In [304]:
head(NYRtrim)

Date               
1 2014-12-21 16:11:00
2 2014-12-21 16:16:00
3 2014-12-21 16:30:00
4 2014-12-21 17:51:00
5 2014-12-21 18:09:00
6 2014-12-21 18:18:00
  Tweet                                                                                                                                           
1 #NewYearsResolution to not put the parking lot ticket directly in my mouth right after getting it from the machine #ew                          
2 I got the hair. I just need to work on the body to pull off Lara Croft next Halloween #NewYearsResolution http://t.co/Ihqh5Z7prx                
3 Self improvement!.. Mentally, physically, and financially. #NewYearsResolution                                                                  
4 I've decided to get pregnant in #2015 #NewYearsResolution                                                                                       
5 All I want for Christmas is to be Carrie Underwood. #NewYearsResolution                                                                         
6 Note to self: "Stop being lazy &amp; get back into hot yoga." I have a monthly pass, so no excuses! #hotyoga #newyearsresolution #healthiswealth
  Category         Topics                                                State
1 Humor            Humor about Personal Growth and Interests Resolutions CA   
2 Health & Fitness Improve my body                                       TX   
3 Personal Growth  Improve my attitude                                   NV   
4 Humor            Humor about Family/Friends/Relationships Resolutions  FL   
5 Personal Growth  Other                                                 SC   
6 Health & Fitness Excersize More                                        NY   
  Region    Gender
1 West      male  
2 South     male  
3 West      male  
4 South     male  
5 South     female
6 Northeast female

# Which region in the United States plans on going to the gym more based on tweets? Target advertising in these regions.


# Total number of Tweets is 4723

In [305]:
str(NYRtrim)

'data.frame':	4723 obs. of  7 variables:
 $ Date    : chr  "2014-12-21 16:11:00" "2014-12-21 16:16:00" "2014-12-21 16:30:00" "2014-12-21 17:51:00" ...
 $ Tweet   : chr  "#NewYearsResolution to not put the parking lot ticket directly in my mouth right after getting it from the machine #ew" "I got the hair. I just need to work on the body to pull off Lara Croft next Halloween #NewYearsResolution http:"| __truncated__ "Self improvement!.. Mentally, physically, and financially. #NewYearsResolution" "I've decided to get pregnant in #2015 #NewYearsResolution" ...
 $ Category: chr  "Humor" "Health & Fitness" "Personal Growth" "Humor" ...
 $ Topics  : chr  "Humor about Personal Growth and Interests Resolutions" "Improve my body" "Improve my attitude" "Humor about Family/Friends/Relationships Resolutions" ...
 $ State   : chr  "CA" "TX" "NV" "FL" ...
 $ Region  : chr  "West" "South" "West" "South" ...
 $ Gender  : chr  "male" "male" "male" "male" ...


# Number of Health and Fitness tweets is 825

In [306]:
table(NYRtrim$Category)


                      Career           Education/Training 
                         123                           87 
Family/Friends/Relationships                      Finance 
                         327                          167 
            Health & Fitness                        Humor 
                         825                          887 
             Personal Growth                Philanthropic 
                        1678                           83 
        Recreation & Leisure Time Management/Organization 
                         461                           85 

In [307]:
NumberOfTweets= 825

# Recode Region

In [309]:
NYRtrim$RegionR [NYRtrim$Region == "West"] < - 1
NYRtrim$RegionR [NYRtrim$Region == "South"] < - 2
NYRtrim$RegionR [NYRtrim$Region == "Northeast"] < - 3
NYRtrim$RegionR [NYRtrim$Region == "Midwest"] < - 4


logical(0)

logical(0)

logical(0)

logical(0)

# Ensure the IV is a factor

In [312]:
str(NYRtrim$RegionR)


 NULL


# Ensure the CV is a factor

In [313]:
str(NYRtrim$Tweet)

 chr [1:4723] "#NewYearsResolution to not put the parking lot ticket directly in my mouth right after getting it from the machine #ew" ...


# Make Tweet numeric

In [314]:
NYRtrim$Tweet <- as.numeric(NYRtrim$Tweet)


Warning message in eval(expr, envir, enclos):
“NAs introduced by coercion”


# Check for Normality

In [315]:
plotNormalHistogram(NYRtrim$RegionR)


Warning message in min(x):
“no non-missing arguments to min; returning Inf”
Warning message in max(x):
“no non-missing arguments to max; returning -Inf”


ERROR: Error in seq.default(min(x), max(x), length = length): 'from' must be a finite number


# Homogeneity of Variance



In [ ]:
leveneTest(Tweet~Region, data=NYRtrim)


Warning message in leveneTest.default(y = y, group = group, ...):
“group coerced to factor.”


ERROR: Error in `contrasts<-`(`*tmp*`, value = contr.funs[1 + isOF[nn]]): contrasts can be applied only to factors with 2 or more levels


# Homogeneity of Regression Slopes


# Sample Size


# ANCOVAs Analysis

In [ ]:
ANCOVA = lm(Category~Tweet +Region*Tweet, data=NYRtrim)
anova(ANCOVA)

ERROR: Error in `contrasts<-`(`*tmp*`, value = contr.funs[1 + isOF[nn]]): contrasts can be applied only to factors with 2 or more levels
